cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -88,51 +107,130 @@ func CreateGceManager(configReader io.Reader) (*GceManager, error) {
 		return nil, err
 	}
 	manager := &GceManager{
-		migs:      make([]*migInformation, 0),
-		service:   gceService,
-		migCache:  make(map[GceRef]*Mig),
-		zone:      zone,
-		projectId: projectId,
+		migs:        make([]*migInformation, 0),
+		gceService:  gceService,
+		migCache:    make(map[GceRef]*Mig),
+		zone:        zone,
+		projectId:   projectId,
+		clusterName: clusterName,
+		mode:        mode,
 		templates: &templateBuilder{
 			projectId: projectId,
 			zone:      zone,
 			service:   gceService,
 		},
 	}
+
+	if mode == ModeGKE {
+		gkeService, err := gke.New(client)
+		if err != nil {
+			return nil, err
+		}
+		manager.gkeService = gkeService
+		err = manager.fetchAllNodePools()
+		if err != nil {
+			glog.Errorf("Failed to fech node pools: %v", err)
+		}
+	}
+
 	go wait.Forever(func() {
 		manager.cacheMutex.Lock()
 		defer manager.cacheMutex.Unlock()
 		if err := manager.regenerateCache(); err != nil {
 			glog.Errorf("Error while regenerating Mig cache: %v", err)
 		}
 	}, time.Hour)
+
 	return manager, nil
 }
 
-// RegisterMig registers mig in Gce Manager.
-func (m *GceManager) RegisterMig(mig *Mig) {
-	m.cacheMutex.Lock()
-	defer m.cacheMutex.Unlock()
+func (m *GceManager) assertGKE() {
+	if m.mode != ModeGKE {
+		panic(fmt.Errorf("This should run only in GKE mode"))
+	}
+}
+
+// Gets all registered node pools
+func (m *GceManager) fetchAllNodePools() error {
+	m.assertGKE()
```

Comment:
Why not just return error if it's not GKE? Or, if we really, really want to guarantee we kill the process if this happens, use Fatalf as we do elsewhere?

Reply:
Done.